diff -urN olsrd-913b11abf7f1.orig/src/linux/net.c olsrd-913b11abf7f1/src/linux/net.c
--- olsrd-913b11abf7f1.orig/src/linux/net.c	2008-12-27 10:41:08.000000000 +0100
+++ olsrd-913b11abf7f1/src/linux/net.c	2008-12-28 22:05:51.000000000 +0100
@@ -678,7 +678,7 @@
 	  OLSR_PRINTF(4, "Found addr: %s:%s:%s:%s:%s:%s:%s:%s\n",
 		      addr6p[0], addr6p[1], addr6p[2], addr6p[3],
 		      addr6p[4], addr6p[5], addr6p[6], addr6p[7]);
-	  inet_pton(AF_INET6, addr6, saddr6);
+	  inet_pton(AF_INET6, addr6, &saddr6->sin6_addr);
 	  rv = 1;
 	  break;
 	}
