diff -Nur olsrd-0.5.6-r2.orig/src/lq_plugin_default_ff.h olsrd-0.5.6-r2/src/lq_plugin_default_ff.h
--- olsrd-0.5.6-r2.orig/src/lq_plugin_default_ff.h	2008-11-26 10:53:16.000000000 +0100
+++ olsrd-0.5.6-r2/src/lq_plugin_default_ff.h	2008-11-26 10:53:16.000000000 +0100
@@ -50,7 +50,8 @@
 
 #define LQ_ALGORITHM_ETX_FF_NAME "etx_ff"
 
-#define LQ_FF_WINDOW 32
+/* 16,32,64 and max. 128 */
+#define LQ_FF_WINDOW 64
 #define LQ_FF_QUICKSTART_INIT 4
 
 struct default_lq_ff {
