Index: boa-0.94.14rc21/Makefile.in
===================================================================
--- boa-0.94.14rc21.orig/Makefile.in	2007-08-08 20:04:19.000000000 -0400
+++ boa-0.94.14rc21/Makefile.in	2007-08-08 20:04:35.000000000 -0400
@@ -20,7 +20,7 @@
 
 mrclean:	clean
 	-(cd src && $(MAKE) $(MFLAGS) mrclean)
-	-(cd docs && $(MAKE)$(MFLAGS) mrclean)
+	-(cd docs && $(MAKE) $(MFLAGS) mrclean)
 	rm -f config.status config.cache config.h config.log 
 	rm -f Makefile *~
 
