diff -urN boa-0.94.13/src/util.c boa/src/util.c
--- boa-0.94.13/src/util.c	2002-07-08 01:22:18.000000000 +0200
+++ boa/src/util.c	2008-04-25 21:56:20.000000000 +0200
@@ -95,14 +95,9 @@
     static char buf[30];
     int time_offset;
 
-    if (use_localtime) {
-        t = localtime(&current_time);
-        time_offset = TIMEZONE_OFFSET(t);
-    } else {
-        t = gmtime(&current_time);
-        time_offset = 0;
-    }
-
+	t = gmtime(&current_time);
+    time_offset = 0;
+    
     p = buf + 29;
     *p-- = '\0';
     *p-- = ' ';

