Index: boa-0.94.13/src/boa.c
===================================================================
--- boa-0.94.13.orig/src/boa.c	2002-07-23 17:50:29.000000000 +0200
+++ boa-0.94.13/src/boa.c	2008-06-29 01:12:31.000000000 +0200
@@ -222,7 +222,7 @@
         /* test for failed-but-return-was-successful setuid
          * http://www.securityportal.com/list-archive/bugtraq/2000/Jun/0101.html
          */
-        if (setuid(0) != -1) {
+        if ((server_uid != 0) && (setuid(0) != -1)) {
             DIE("icky Linux kernel bug!");
         }
     } else {
