Index: boa-0.94.13/src/compat.h
===================================================================
--- boa-0.94.13.orig/src/compat.h	2002-07-23 17:50:29.000000000 +0200
+++ boa-0.94.13/src/compat.h		2008-09-13 21:55:31.000000000 +0200
@@ -70,8 +70,8 @@
 #endif
 
 #ifdef INET6
-#define SOCKADDR sockaddr_storage
-#define S_FAMILY __s_family
+#define SOCKADDR sockaddr_in6
+#define S_FAMILY sin6_family
 #define SERVER_AF AF_INET6
 #else
 #define SOCKADDR sockaddr_in
 #define S_FAMILY sin_family
