diff -ur lua-5.1.3/etc/lua.pc lua-5.1.3-new/etc/lua.pc
--- lua-5.1.3/etc/lua.pc        2008-01-11 14:12:59.000000000 +0100
+++ lua-5.1.3-new/etc/lua.pc    2008-05-18 22:09:24.000000000 +0200
@@ -8,7 +8,7 @@
 R= 5.1.3

 # grep '^INSTALL_.*=' ../Makefile | sed 's/INSTALL_TOP/prefix/'
-prefix= /usr/local
+prefix= /usr
 INSTALL_BIN= ${prefix}/bin
 INSTALL_INC= ${prefix}/include
 INSTALL_LIB= ${prefix}/lib
diff -ur lua-5.1.3/Makefile lua-5.1.3-new/Makefile
--- lua-5.1.3/Makefile  2008-05-18 22:06:55.000000000 +0200
+++ lua-5.1.3-new/Makefile      2008-05-18 22:09:13.000000000 +0200
@@ -12,7 +12,7 @@
 # doc directory.) You may want to make these paths consistent with LUA_ROOT,
 # LUA_LDIR, and LUA_CDIR in luaconf.h (and also with etc/lua.pc).
 #
-INSTALL_TOP= /usr/local
+INSTALL_TOP= /usr
 INSTALL_BIN= $(INSTALL_TOP)/bin
 INSTALL_INC= $(INSTALL_TOP)/include
 INSTALL_LIB= $(INSTALL_TOP)/lib
diff -ur lua-5.1.3/src/luaconf.h lua-5.1.3-new/src/luaconf.h
--- lua-5.1.3/src/luaconf.h     2008-05-18 22:07:11.000000000 +0200
+++ lua-5.1.3-new/src/luaconf.h 2008-05-18 22:09:50.000000000 +0200
@@ -95,7 +95,7 @@
        ".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"

 #else
-#define LUA_ROOT       "/usr/local/"
+#define LUA_ROOT       "/usr/"
 #define LUA_LDIR       LUA_ROOT "share/lua/5.1/"
 #define LUA_CDIR       LUA_ROOT "lib/lua/5.1/"
 #define LUA_PATH_DEFAULT  \
