diff -urb axTLS.orig/httpd/Makefile axTLS/httpd/Makefile
--- axTLS.orig/httpd/Makefile	2008-11-09 00:58:44.000000000 +0000
+++ axTLS/httpd/Makefile	2008-11-09 01:29:55.000000000 +0000
@@ -95,7 +95,7 @@
 ifndef CONFIG_PLATFORM_WIN32
 
 $(TARGET): $(OBJ) $(AXTLS_HOME)/$(STAGE)/libaxtls.a
-	$(LD) $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
 ifdef CONFIG_STRIP_UNWANTED_SECTIONS
 	strip --remove-section=.comment $(TARGET)
 endif
